src/Hamiltonians/TwoBodyCorrelation.jl

```diff
@@ -0,0 +1,74 @@
+# Methods that need to be implemented:
+#   •  starting_address(::AbstractHamiltonian) - not needed
+"""
+    G2Correlator(d::Int)
+
+Implements the two-body correlation operator for computing the correlation
+between two particles in two different components of a Fock states encoded
+in momentum space. The variable `d` is the distance between the two particles
+which can take `Integer` values from `0` to `M` (the number of modes).
```

Comment:
```suggestion
    G2Correlator(d::Int) <: AbstractHamiltonian{ComplexF64}

Two-body correlation operator representing the inter-component density-
density correlation at distance `d` of a two component system 
in a momentum-space Fock-state basis. 
```
I think we don't need to restrict the range of `d` here, because it will be taken modulo `M` anyway.